[이승헌] sprint 1

heony704 opened the pull request:
## 요구사항

디자인: [판다마켓 Figma](https://www.figma.com/file/IVkRlYWHY74QlgmxqA99Ym/%EC%8A%A4%ED%94%84%EB%A6%B0%ED%8A%B8-%EB%AF%B8%EC%85%98?type=design&node-id=55-1511&mode=design&t=UPQqQU1ojryVqNNl-0)

### 기본

- [x] 랜딩 페이지의 url path는 루트(‘/’)로 설정합니다.
- [x] title은 “판다마켓”로 설정합니다.
- [x] 화면의 너비가 1920px 이상이면 하늘색 배경색은 너비를 꽉 채우도록 채워지고, 내부 요소들의 위치는 고정되고, 여백만 커지도록 합니다.
- [x] 화면의 너비가 1920px 보다 작아질 때, “판다마켓” 로고의 왼쪽 여백 200px“로그인" 버튼의 오른쪽 여백 200px이 유지되고, 화면의 너비가 작아질수록 두 요소간 거리가 가까워지도록 설정합니다.
- [x] 클릭으로 기능이 동작해야 하는 경우, 사용자가 클릭할 수 있는 요소임을 알 수 있도록 CSS 속성 cursor: pointer 로 설정합니다.
- [x] “판다마켓” 클릭 시 루트 페이지(‘/’)로 이동합니다.
- [x] '로그인'버튼 클릭 시 로그인 페이지(‘/login’)로 이동합니다 (빈 페이지)
- [x] “구경하러가기”버튼 클릭 시(’/items’)로 이동합니다.(빈 페이지)
- [x] “Privacy Policy”, “FAQ”는 클릭 시 각각 Privacy 페이지(‘/privacy’), FAQ 페이지(‘/faq’)로 이동합니다.(모두 빈 페이지)
- [x] 페이스북, 트위터, 유튜브, 인스타그램 아이콘을 클릭 시 각각의 홈페이지로 새로운 창이 열리면서 이동합니다.

### 심화

- [x] palette에 있는 color값들을 css 변수로 등록하고 사용해 주세요.

## 주요 변경사항

- 랜딩 페이지 추가

## 스크린샷

🚀 배포 페이지: [heony704-pandamarket-1.netlify.app](https://heony704-pandamarket-1.netlify.app/)

![판다마켓_랜딩페이지](https://github.com/codeit-bootcamp-frontend/6-Sprint-Mission/assets/36994104/5b8fff41-b3f7-4c0e-b6c6-2509950273bc)

## 멘토에게

- 최대한 Figma와 비슷하게 만들면서 코드를 깔끔하게 작성하려고 노력했습니다.
- 버튼을 누르면 다른 페이지로 이동하는 기능을 구현할 때
  button 태그를 a 태그로 감싸는 대신 button에 onclick으로 이동하도록 해봤는데 뭐가 더 나은지 잘 모르겠습니다.
- 시맨틱 태그 활용하기가 어렵습니다. 
  멘토님은 시맨틱 태그를 어떻게 활용하시는지 궁금합니다.

Commit:
Feat: 기본 구조 만들기

- assets 폴더에 img, svg 파일 관리
- html 구조 만듦
- css 파일 만들고 색상 변수 및 css reset 코드 추가

Commit:
Design: 메인페이지

Commit:
Feat: button, p 클릭 시 페이지 이동

- 판다마켓 로고 클릭 시 ('/')로 이동
- 로그인 버튼 클릭 시 ('/login')로 이동
- 구경하러가기 버튼 클릭 시 ('/items')로 이동
- Privacy Policy 클릭 시 ('/privacy')로 이동
- FAQ 클릭 시 ('/faq')로 이동
- 페이스북, 트위터, 유튜브, 인스타 아이콘 클릭 시 홈페이지 창 켜짐

Commit:
Design: footer 링크 색 변경

Commit:
Design: 버튼 호버 및 클릭시의 색상 추가

- 색상 변수 추가
- 버튼 호버 및 클릭 시 색 변하도록 수정

kich555 commented on style.css:
```diff
@@ -0,0 +1,23 @@
+  --gray100: #e8ebed;
+  --gray50: #f7f7f8;
+
+  --blue: #3692ff;
```
음... `--blue` 라는 변수명보다는 `--primary` 가 더 괜찮을것 같은데 어떻게 생각하세요? 
아래 피그마 파일을 간단하게 보면 디자인을 관통하는 메인 컬러를 명시한거지 그냥 단순히 파란색을 명시했다고는 잘 안보여서요 ㅎ

<img width="763" alt="image" src="https://github.com/codeit-bootcamp-frontend/6-Sprint-Mission/assets/79491683/c0881eb2-2ea3-4eb8-8033-0e35cb0af169">

kich555 replied:
흠; 다시보니 blue라고 적혀있긴 하네요. 만약 현업에서 이럴경우 저는 저 파란색이 디자인 main을 관통하는 컬러인지 아닌지 디자이너한테 한번 물어보고 결정할 것 같아요!

kich555 commented on style.css:
```diff
@@ -0,0 +1,23 @@
+  --gray50: #f7f7f8;
+
+  --blue: #3692ff;
+  --red: #f74747;
```
요것도 마찬가지
<img width="270" alt="image" src="https://github.com/codeit-bootcamp-frontend/6-Sprint-Mission/assets/79491683/2cb07b44-d816-4801-beb8-6f8461422c8f">

`red` 긴 하지만 피그마 디자인에 해당 색상이 사용된 부분이 보이지 않는거로 봐선 
왠지 저는 나중에 에러 UI를 추가할때 사용한 컬러인건가? 하고 한번쯤 의심해볼 것 같아요
마찬가지로 디자이너한테 문의한 후 `red` `error`중 하나로 이름을 결정할 것 같습니다.

kich555 replied:
추가로 위와 같은경우에는 
```
/* TODO: 해당 컬러에대한 이름이 아직 fix되지 않음. 디자이너와 상의하고 결정할것 */
--red: #f74747
```

와 같이 주석을 남겨둘것 같습니다!

kich555 commented on style.css:
```diff
@@ -0,0 +1,23 @@
+  --red: #f74747;
+}
+
+* {
```
혹시 reset.css , normalize css등에 대해 안배웠나요? 
https://www.daleseo.com/css-normalize-reset/ 글 한번 읽어보는걸 추천드립니다!

kich555 commented on index.html:
```diff
@@ -0,0 +1,90 @@
+  </head>
+  <body>
+    <header>
+      <div>
```
혹시 header tag안에 굳이 div를 하나 더 넣으신 이유가 있으신가요? 
이부분만 볼땐 제 눈에는 해당 이유가 보이질 않아서요!

kich555 commented on index.html:
```diff
@@ -0,0 +1,90 @@
+    <header>
+      <div>
+        <img class="logo" src="/assets/logo.png" alt="판다마켓 로고" />
+        <button class="small">로그인</button>
```
`<img class="logo"`, `<button class="small"` 같이 class들 선언해 주셨는데 관련한 css파일이 안보이는것 같습니다 한번 확인해주실 수 있으실까요? 
그리고 클래스명 관련해서 피드백 드리고싶은게 있는데요, 
이거 지금 하나의 html에 css 넣어 사용중인건데...
small, logo... 클래스명들이 보다 명시적이면 좋지 않을까요?  이런 네이밍은 금방 class name충돌 일으킬 것 같은데..어떻게 생각하세요? 
가령 `logo-img`, `small-button`같이 보다 명시적이게 네이밍을 가져가보는게 더 좋지 않을까요?

kich555 commented on index.html:
```diff
@@ -0,0 +1,90 @@
+      </div>
+    </header>
+    <main>
+      <div class="banner">
```
밑에 보면 `<main>` 안에 `<section>` 들로 감싸여져 있던데.. 왜 이 친구는 `<div>` 태그로 하셨는지 여쭤봐도 될까요?

kich555 commented on index.html:
```diff
@@ -0,0 +1,90 @@
+          </h1>
+          <button class="large">구경하러 가기</button>
+        </div>
+        <img src="/assets/img_home_top.png" alt="판다가 인사하는 모습" />
```
`alt` 속성 꼼꼼히 넣어주시는것 좋습니다 !! 👍👍👍👍

kich555 commented on index.html:
```diff
@@ -0,0 +1,90 @@
+      </section>
+      <section>
+        <img src="/assets/img_home_02.png" alt="상품을 검색하는 모습" />
+        <div class="desc">
```
마찬가지로 class명 좀더 신경써보기...!

kich555 commented on index.html:
```diff
@@ -4,6 +4,12 @@
+      rel="stylesheet"
+      as="style"
+      crossorigin
+      href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.9/dist/web/variable/pretendardvariable.min.css"
```
여기에 추가되었군요 ㅎ 이것도 initial setting commit에 포함되었어야하지 않았나 생각해봐요 ㅎ

kich555 commented on index.html:
```diff
@@ -14,76 +20,88 @@
     <main>
       <div class="banner">
-        <div>
+        <div class="container contents">
```
class 조합해서 사용하는것 좋습니다! 👍👍👍

kich555 commented on index.html:
```diff
@@ -14,76 +20,88 @@
-        <img src="/assets/img_home_03.png" alt="여러 상품이 나열된 모습" />
-        <div class="desc">
-          <p>Register</p>
+      <article class="container">
```
여기서부터 prettier같은걸 사용하셨나요? formatting 규칙이 변경된것 같네요?

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
   margin: 0;
   padding: 0;
+
+  font-family: "Pretendard Variable", Pretendard, -apple-system,
```
이것도 initial setting 혹은 "font 추가" 와 같은 커밋으로 따로 분리되었다면 좋았을것 같네요 ㅎ

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+}
+
+header > div {
+  max-width: 1920px;
```
1920같은 값은 제가 봤을때 viewport 관련한 값 같은데요 단순히 header > div같이 들어가기보단 반응형 관련된 다른 클래스 같은거로 분리하는건 어떨까요?

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+  margin: 0 auto;
+  padding: 10px 200px;
+  height: 100%;
+  display: flex;
```
음...그리고 display > position > boxmodel -> content관련속성 > text관련 속성 과 같은 순서로 배치해두는 컨벤션이 나중에 가독성 좋더라구요! 
필수는 아닌데 ㅎ 이런 컨벤션 정도는 지켜두면 나중에 다른 개발자들이 좋아할 것 같아요 !

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+
+button {
+  border: 0;
+  background: var(--blue);
```
👍👍👍

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+
+.logo {
+  height: 100%;
+  cursor: pointer;
```
취향 차이긴 한데요. 저같으면 pointer 라는 이름의 class를 만들고 html에서 사용할때 해당 class를 추가하는 식으로 사용할 것 같아요 
뭐 다른 요소들도 마찬가지긴한데 가령
```
flex-center {
display:flex;
align-items:center;
justify-content:center;
}
```
이런식으로 자주 중복해서 사용하는 코드들을 모듈화 해두면 나중에 생산성이 올라가지 않을까요?

kich555 replied:
(나중에 tailwind배우시게 될텐데 그친구가 위에 설명한 식으로 되어있을거에요 ㅎㅋ)

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+  cursor: pointer;
+}
+
+button.small {
```
여기서 뭔가 느끼셨을지도 모르겠는데 실은 이런 수치값들은 보통 4의배수, 혹은 2의배수의 성향을 띕니다. 가끔 이게 지켜지지않은 값을 보면 한번씩 디자이너한테 더블체크를 해보는것도 좋을 것 같습니다
**(물론 당연하게도 이건 디자이너가 이런 컨벤션을 지키며 디자인하는 디자이너에 한해서 더블체크해야합니다..!!)**

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+}
+
+.banner {
+  background: #cfe5ff;
```
이 값은 따로 변수화 해두지 않은 값인건가요? 혹시 변수로 지정해둔 hex값인지 한번만 더 체크해주세요! figma자체에서 없는거라면 ok!

kich555 commented on style.css:
```diff
@@ -16,8 +17,168 @@
+  width: 996px;
+  position: absolute;
+  bottom: 0;
+  right: -135px;
```
right: -135px; 이 코드 혹시 브라우저 사이즈가 줄어들어도 위와같은 간격만큼 떨어지는게 맞는건가요? 
그리고 
<img width="620" alt="image" src="https://github.com/codeit-bootcamp-frontend/6-Sprint-Mission/assets/79491683/a643b779-6169-469a-a7e9-989bd227f787">

여기 보면 배너 텍스트가 왼쪽 브라우저 끝단이랑 너무 딱붙어있는데 이거 줄어들면서 간격이 사라졌거든요? 디자인도 혹시 의도된건지 한번만 확인 부탁드립니다!

kich555 commented on index.html:
```diff
@@ -14,8 +14,12 @@
+        <a class="logo" href="/">
+          <img src="/assets/logo.png" alt="판다마켓 로고" />
+        </a>
+        <button class="small" onclick="location.href='/login.html'">
```
이게 UI는 버튼인데 기능은 a 태그라서 이렇게 해두신거죠? 
저...라면 이럴때 button태그 안쓰고 a태그 사용할것 같아요 
뭐가 정답인지는....저도 잘 모르겠네요 다만 시멘틱적인 구조상은 a태그를 사용하는게 맞아보입니다 ㅎ

kich555 commented on style.css:
```diff
@@ -173,12 +180,20 @@ footer .links {
   gap: 30px;
 }
 
+footer .links a:visited {
```
굳 👍👍👍

kich555 commented on style.css:
```diff
@@ -9,8 +9,13 @@
 
-  --blue: #3692ff;
-  --red: #f74747;
+  --primary700: #1251aa;
```
오 이미 변경하셨네요 ㅎㅎ 굳굳 👍👍

kich555 reviewed (commented):
전체적으로 정말 잘해주셨어요!! 
매운맛으로 해달라하셔서 굳이 안잡아도 될 포인트들도 많이 잡았습니다 ㅎ 
딱 현업에서 마주쳤을때의 정도로 리뷰한것같아요 
고생 많으셨습니다 !!! 

하나 추가로 말씀드리고싶은건 
[Feat: button, p 클릭 시 페이지 이동](https://github.com/codeit-bootcamp-frontend/6-Sprint-Mission/pull/44/commits/ba8c9ca272d251ce2fcc587a05db1711404b5752)
 커밋메세지를 그냥 단순하게 link 기능 적용 같이..? 해두는건 어떨까요?

kich555 commented on style.css:
```diff
@@ -0,0 +1,220 @@
+  height: 160px;
+}
+
+footer > div {
```
이런 부분들....만약에 html구조가 바뀌게되면 css가 확 깨질수도있을것 같아서 저는 개인적으로 nesting되었더라도 tag자체에 스타일 주는건 별로 선호하는 방식은 아닙니다 ㅎ 왜냐면 html파일 만 봐서는 해당 div에 아래와 같은 스타일이 적용되어있는지 없는지 전혀 알 수 없기 때문이에요 
내가 알고있다면 문제가 되지 않지만 협업하는 과정속에는 위와같이 **컨텍스트를 모르면 실수가 나기 좋은 코드** 는 최대한 지양해주시는게 좋습니다 !

Commit:
Design: css 클래스 이름 변경

기존
- 클래스 이름을 사용하지 않고 태그 이름으로 스타일 적용

수정 후
- css 클래스를 만들어 클래스 이름을 통해 스타일 적용
- 스타일이 태그 이름에 의존하지 않게 됨

Commit:
Style: css 작성 순서 변경

기존
- css 작성 순서가 중구난방이라 가독성 떨어짐

수정 후
- display, position, box model, content, text 순서로 작성

Commit:
Refactor: 버튼의 link 연결 방식 변경

기존
- button 태그의 onclick 이용

수정 후
- a 태그의 href 이용

- 다른 페이지로 이동하는 버튼이므로 a 태그가 좀 더 목적에 맞다고 판단함

Commit:
Refactor: 시맨틱 태그 적용

기존
- 배너에 div 태그 사용

수정 후
- 배너에 section 태그 사용

Commit:
Refactor: css 파일 분리

- 유지보수하기 쉽도록 css 파일 분리

kich555 commented on style.css:
```diff
@@ -52,15 +52,11 @@ header > div {
 }
 
-.logo {
+.h-100 {
```
음...이건 너무 줄인것 같아요
그리고 보통 부트스트랩이나 테일윈드같은 라이브러리의 코드 컨벤션에서는 `h-100`하면 보통 `heigth: 100px`을 연상시키기 때문에 다른 이름으로 바꿔보시는건 어떨까요?

그냥 `height: 100%` 만 주고싶으면 `full-height` 같은 클래스 명을 두는게 재사용성을 높히는 측면에서는더 좋지 않을까요?

kich555 commented on styles/common.css:
공통적으로 사용하는 클래스들 이렇게 따로 파일 나눠두신거 좋아요!!! 굿굿 👍👍👍👍

kich555 reviewed (approved):
전반적으로 리뷰해 드린 부분들 다 깔끔히 지켜주신 것 같아서 감사합니다!!!
고생 많으셨습니다!!

Commit:
Refactor: css 클래스 이름 개선

기존
- h-100 이라는 클래스 이름은 이해하기 어려움

수정 후
- full-heigth 라는 이름으로 height:100% 라는 의미가 전달되도록 수정
